bootstrap_actions/cloudwatch.sh

```diff
@@ -25,107 +25,111 @@ cat > /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.json <<CWAGEN
           },
           {
             "file_path": "/var/log/adg/install-boto3.log",
-            "log_group_name": "${cwa_bootstrap_loggrp_name}",
+            "log_group_name": "$${cwa_bootstrap_loggrp_name}",
             "log_stream_name": "install-boto3.log",
             "timezone": "UTC"
           },
           {
             "file_path": "/var/log/adg/hive-tables-creation.log",
-            "log_group_name": "${cwa_steps_loggrp_name}",
+            "log_group_name": "$${cwa_steps_loggrp_name}",
             "log_stream_name": "hive-tables-creation.log",
             "timezone": "UTC"
           },
           {
             "file_path": "/var/log/adg/create-hive-tables.log",
-            "log_group_name": "${cwa_steps_loggrp_name}",
+            "log_group_name": "$${cwa_steps_loggrp_name}",
             "log_stream_name": "create-hive-tables.log",
             "timezone": "UTC"
           },
           {
             "file_path": "/var/log/adg/generate-analytical-dataset.log",
-            "log_group_name": "${cwa_steps_loggrp_name}",
+            "log_group_name": "$${cwa_steps_loggrp_name}",
             "log_stream_name": "generate-analytical-dataset.log",
             "timezone": "UTC"
           },
           {
             "file_path": "/var/log/hadoop-yarn/containers/application_*/container_*/stdout**",
-            "log_group_name": "${cwa_yarnspark_loggrp_name}",
+            "log_group_name": "$${cwa_yarnspark_loggrp_name}",
             "log_stream_name": "spark-stdout.log",
             "timezone": "UTC"
           },
           {
             "file_path": "/var/log/hadoop-yarn/containers/application_*/container_*/stderr**",
-            "log_group_name": "${cwa_yarnspark_loggrp_name}",
+            "log_group_name": "$${cwa_yarnspark_loggrp_name}",
             "log_stream_name": "spark-stderror.log",
             "timezone": "UTC"
           },
           {
             "file_path": "/var/log/hadoop-yarn/yarn-yarn-nodemanager**.log",
-            "log_group_name": "${cwa_yarnspark_loggrp_name}",
+            "log_group_name": "$${cwa_yarnspark_loggrp_name}",
             "log_stream_name": "yarn_nodemanager.log",
             "timezone": "UTC"
           }
         ]
       }
     },
-    "log_stream_name": "${cwa_namespace}",
+    "log_stream_name": "$${cwa_namespace}",
     "force_flush_interval" : 15
   }
 }
 CWAGENTCONFIG
 
+%{ if emr_release == "5.29.0" ~}
 # Download and install CloudWatch Agent
-curl https://s3.${AWS_DEFAULT_REGION}.amazonaws.com/amazoncloudwatch-agent-${AWS_DEFAULT_REGION}/centos/amd64/latest/amazon-cloudwatch-agent.rpm -O
+curl https://s3.$${AWS_DEFAULT_REGION}.amazonaws.com/amazoncloudwatch-agent-$${AWS_DEFAULT_REGION}/centos/amd64/latest/amazon-cloudwatch-agent.rpm -O
 rpm -U ./amazon-cloudwatch-agent.rpm
 # To maintain CIS compliance
 usermod -s /sbin/nologin cwagent
 
 start amazon-cloudwatch-agent
+%{ else ~}
+systemctl start amazon-cloudwatch-agent
+%{ endif ~}
```

Comment:
This doesn't work.
Should be something like...
`sudo /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a fetch-config -m ec2 -c ssm:${cloudwatch_agent_config_ssm_parameter} -s`